apps/damap-frontend/src/app/components/layout/layout.component.ts

```diff
@@ -31,6 +45,45 @@ export class LayoutComponent implements OnInit {
     const browserLang = this.translate.getBrowserLang();
     this.translate.use(browserLang?.match(/en|de/) ? browserLang : 'en');
     this.lang = this.translate.currentLang.toUpperCase();
+
+    this.observer.observe([Breakpoints.Handset]).subscribe(result => {
+      this.isSmallScreen = result.matches;
+      if (this.sidenav) {
+        if (this.isSmallScreen) {
+          this.isCollapsed = true;
+          this.sidenav.mode = 'side';
+          this.sidenav.open();
+        } else {
+          this.isCollapsed = false;
+          this.sidenav.mode = 'side';
+          this.sidenav.open();
+        }
+      }
+    });
+  }
+
+  ngAfterViewInit(): void {
+    this.checkScreenSize();
+  }
+
+  onResize() {
+    this.checkScreenSize();
+  }
+
+  checkScreenSize() {
+    this.isSmallScreen = window.innerWidth < 1024;
+    if (this.sidenav) {
+      if (this.isSmallScreen) {
+        this.isCollapsed = true;
+        this.sidenav.mode = 'side';
+        this.sidenav.open();
+      } else {
+        this.isCollapsed = false;
+        this.sidenav.mode = 'side';
+        this.sidenav.open();
+      }
+    }
+    this.cdr.detectChanges();
   }
```

Comment:
Duplicated logic from https://github.com/tuwien-csd/damap-frontend/pull/270/files#diff-ba993de46fd16faf82832c99115f225b81031881a7080216f02663de98be5635R49-R62

Can this be removed?